### DISCUSSION TITLE
Vite SSR breaking app if used with useHydrationAtoms

### DISCUSSION THREAD

**Discussion by spacepolice10** [pmndrs/jotai#2564] [Q&A]
Hi there, 

Not sure if it is Jotai or Vite related problem and can't figure it out on my own. I have a Remix application that should hydrate atoms with data from server. But I encountered an error "TypeError: Invalid value used in weak set". 

The code is here: https://codesandbox.io/p/devbox/romantic-villani-wrswh7

If I add a `<Suspense />` tag, it'll break hydration and spam me with mismatch errors. And without `<Suspense />` it just fails to render.

**Comment by dai-shi** [comment]
There's only one place useHydrateAtoms uses `WeakSet`. I assume you are trying to pass atom object from the server to the client. Atoms should exists on the client only. "use client"

**Reply by spacepolice10** [reply]
So it is probably bug in Vite :( I actually marked file where atom is initialized as `client` only so it should not exist on server (should be undefined). Ok, I'll try to find out what might went wrong. Thank you

**Comment by spacepolice10** [comment]
Forgot to close the issue. I found out that it was a problem with atoms created client-side only (somehow). At the stage of hydrating atoms were `undefined`. So I recommend anyone with the same problem to not use `clientOnly$` function or `.client` extension in places where `atoms` declared. Cheers
